services/models/health.go

```diff
@@ -18,3 +21,38 @@ type DeploymentStatus struct {
 	Replicas          int32  `json:"replicas"`
 	AvailableReplicas int32  `json:"available"`
 }
+
+// RequestHealth holds several stats about recent request errors
+type RequestHealth struct {
+	RequestCount      float64 `json:"requestCount"`
+	RequestErrorCount float64 `json:"requestErrorCount"`
+	Fetched           bool    `json:"-"`
+}
+
+// FillDeploymentStatusesIfMissing sets DeploymentStatuses if necessary
+func (in *Health) FillDeploymentStatusesIfMissing(supplier func() []DeploymentStatus) {
+	if !in.DeploymentsFetched {
```

Comment:
I wonder, it is not enough to check a non nil of DeploymentStatuses ?

Reply:
It wouldn't work, because it's an array so its default value is an empty array. So the "not fetched" state and "no deployment" state could not be distinguished.